pkg/controller/certificaterequests/sync.go

```diff
@@ -161,19 +164,38 @@ func (c *Controller) Sync(ctx context.Context, cr *cmapi.CertificateRequest) (er
 	return nil
 }
 
-func (c *Controller) updateCertificateRequestStatusAndAnnotations(ctx context.Context, old, new *cmapi.CertificateRequest) (*cmapi.CertificateRequest, error) {
+func (c *Controller) updateCertificateRequestStatusAndAnnotations(ctx context.Context, old, new *cmapi.CertificateRequest) error {
 	log := logf.FromContext(ctx, "updateStatus")
 
 	// if annotations changed we have to call .Update() and not .UpdateStatus()
 	if !reflect.DeepEqual(old.Annotations, new.Annotations) {
 		log.V(logf.DebugLevel).Info("updating resource due to change in annotations", "diff", pretty.Diff(old.Annotations, new.Annotations))
-		return c.cmClient.CertmanagerV1().CertificateRequests(new.Namespace).Update(ctx, new, metav1.UpdateOptions{})
+		return c.updateOrApply(ctx, new)
```

Comment:
Note: not a change made in this PR, but I wonder if, in a scenario where both annotations and status have been modified, returning here after applying _just_ the annotations could not result in an incorrect state.

Reply:
Indeed, CertificateRequest controllers should not be attempting to update _both_ the object meta and status in the same reconcile.